python/pyspark/ml/tests.py

```diff
@@ -1027,6 +1027,21 @@ def test_storage_levels(self):
         self.assertEqual(als.getFinalStorageLevel(), "DISK_ONLY")
         self.assertEqual(als._java_obj.getFinalStorageLevel(), "DISK_ONLY")
 
+    def test_unknown_strategy(self):
```

Comment:
I agree with @sethah that testing the setting/transferring of the Params here is not needed and would be better to leave to the wrapper class to test.  Also, `_java_obj` is supposed to be a "private" variable and maybe not good for the end user to think it's ok to use directly.

Perhaps a more useful test could be what happens if the user sets the Param to an unsupported value, e.g.
```py
als.setUnknownStrategy("duh")
```
Is the expected behaviour that it would fail when `fit` is called?  If so, I don't think that's great, but that's more of a problem of the wrapper class anyway.